src/monsters.cpp

```diff
@@ -186,6 +186,121 @@ void MonsterType::createLoot(Container* corpse)
 	corpse->startDecaying();
 }
 
+void MonsterType::clone(MonsterType* root, MonsterType* child, const std::string type) {
+	if (type == "attacks") {
+		if (!child->spellAttackList.empty()) {
+			for (spellBlock_t attacks : child->spellAttackList) {
+				root->spellAttackList.push_back(attacks);
+			}
+		}
+	} else if (type == "defenses") {
+		if (!child->spellDefenseList.empty()) {
+			for (spellBlock_t defenses : child->spellDefenseList) {
+				root->spellDefenseList.push_back(defenses);
```

Comment:
You would need to create an exact copy of the spell object that's pointed to by the spell ptr in the spell block (or introduce a ref counting scheme to Spell objects and use a custom smart pointer). Also I think you should look at this PR #1251. I've made a bunch of improvements there you might also look at. If you decide to introduce ref counting to spells, please let me know, I'll help you do it properly.